frontend/src/lib/lemon-ui/LemonCalendar/LemonCalendarSelect.tsx

```diff
@@ -1,20 +1,101 @@
 import { IconX } from '@posthog/icons'
 import { dayjs } from 'lib/dayjs'
 import { LemonButton, LemonButtonWithSideActionProps, SideAction } from 'lib/lemon-ui/LemonButton'
-import { LemonCalendar } from 'lib/lemon-ui/LemonCalendar/LemonCalendar'
-import { useState } from 'react'
+import { GetLemonButtonTimePropsOpts, LemonCalendar } from 'lib/lemon-ui/LemonCalendar/LemonCalendar'
+import { useEffect, useMemo, useRef, useState } from 'react'
 
 import { Popover } from '../Popover'
 
+function timeDataAttr({ unit, value }: GetLemonButtonTimePropsOpts): string {
+    return `${value}-${unit}`
+}
+
+export function getTimeElement(
+    parent: HTMLElement | null,
+    props: GetLemonButtonTimePropsOpts
+): HTMLDivElement | undefined | null {
+    return parent?.querySelector(`[data-attr="${timeDataAttr(props)}"]`)
+}
+function scrollToTimeElement(
+    calendarEl: HTMLDivElement | null,
+    props: GetLemonButtonTimePropsOpts,
+    skipAnimation: boolean
+): void {
+    getTimeElement(calendarEl, props)?.scrollIntoView({
+        block: 'start',
+        inline: 'nearest',
+        behavior: skipAnimation ? ('instant' as ScrollBehavior) : 'smooth',
```

Comment:
Need specific typing to fix https://github.com/microsoft/TypeScript/issues/47441